### DISCUSSION TITLE
Consider Ruff for formatting/linting

### DISCUSSION THREAD

**Discussion by keikoro** [Zulko/moviepy#2342] [Ideas]
Our [formatting/linting workflow](https://github.com/Zulko/moviepy/blob/49e3f2af039d977a0fe80d2a5ad3afa5334b9009/.github/workflows/formatting_linting.yml) currently uses 3 different tools, black, flake8 and isort, which could be replaced by [Ruff](https://github.com/astral-sh/ruff).

I didn't want to introduce new tools when I reworked the workflow before the v2 release, but would like to suggest this now, also because's there's a [pyproject.toml](https://github.com/Zulko/moviepy/blob/49e3f2af039d977a0fe80d2a5ad3afa5334b9009/pyproject.toml) file now which can be used to store [tool config](https://docs.astral.sh/ruff/configuration/).

**Comment by OsaAjani** [comment]
I would be ok with it, but I've never used Ruff, so I can't help on this one. If you want to go forward with it make sure to also update the doc. Using only one tool would indeed be nice, even nicer would be an auto running of the tool as some pre-commit hook ;)
